packages/components/src/components/table/table.tsx

```diff
@@ -0,0 +1,79 @@
+import { Component, Prop, h, Host, Element, State } from '@stencil/core';
+import { CssClassMap } from '../../utils/utils';
+import classNames from 'classnames';
+import { styles } from './table.styles';
+import { CssInJs } from '../../utils/css-in-js';
+import { StyleSheet } from 'jss';
+import Base from '../../utils/base-interface';
+
+@Component({
+  tag: 'scale-table',
+  shadow: true,
+})
+export class Table implements Base {
+  @Element() hostElement: HTMLElement;
+  /** (optional) Tag class */
+  @Prop() customClass?: string = '';
+  /** (optional) Visual variant */
+  @Prop() variant?: 'regular' | 'compressed' = 'regular';
+  /** (optional) Injected jss styles */
+  @Prop() styles?: StyleSheet;
+  /** decorator Jss stylesheet */
+  @CssInJs('Table', styles) stylesheet: StyleSheet;
+  /** object of the slots in use */
+  @State() slots: { header?: Element; table?: Element } = {};
+
+  componentWillLoad() {}
+
+  componentWillUpdate() {}
+
+  render() {
+    // on initial render
+    if (!this.slots.table) {
+      // build object of slots
+      // @ts-ignore - fromEntries should be fine here
+      this.slots = Object.fromEntries(
+        Array.from(this.hostElement.children).map(v => [v.slot, v])
```

Comment:
Please avoid shortcuts 😉: instead of `...map(v => [v.slot, v])` better do `...map(child => [child.slot, child])`